syntax = "proto3";

import "vgeodb_common.proto";

package VgeoDB;

option java_package = "vgeodb.placevector";
option java_outer_classname = "prdb";


message PlaceVectorMeta_t {
	uint64 un64UniqueID			= 1;
	uint64 un64KeyframeIndex	= 2;
	uint64 un64GroupID			= 3;
	uint64 un64Timestamp		= 4;

	VgeoDB.common.XYZ64_t stAvgLocation	= 5;
	VgeoDB.common.Orientation32_t stOrientaton	= 6;

	repeated VgeoDB.common.XYZ64_t arstLocationSet = 7;

	VgeoDB.common.XYZ32_t stDirectionVector = 8;

	repeated uint64 arun64UniqueIDofKeypoint = 9 [packed=true];
	repeated uint64 arun64Belong3DPointIndex = 10 [packed=true];

	float fSimilarity = 11;
}

message NetVLADDesc_t {
	enum eDescType {
		VLAD_4096_v1 = 0;
	}
	repeated float arfNetVLADDescriptor = 1 [packed=true];

	eDescType eType = 2;
}

message GlobFeatDesc_t {
	enum eDescType {
		GLOB_8192_v1 = 0;
	}
	repeated uint32 arunGlobFeatDescriptor = 2 [packed=true];
	eDescType eType = 3;
}

message PlaceVectorDesc {
	oneof uDescriptor {
		NetVLADDesc_t	stNetVLADDescriptor = 1;
		GlobFeatDesc_t	stGlobFeatDescriptor = 2;
	}
}

message PlaceVector {
	PlaceVectorMeta_t stMeta = 1;
	PlaceVectorDesc uDescriptor = 2;
}

message PlaceVectorSet {
	repeated PlaceVector arstPlaceVector = 1;
}
